CodeJam.Main/Targeting/TypeExtensions.cs

```diff
@@ -145,121 +145,6 @@ internal static class TypeExtensions
-			[NotNull] this Type type,
-			BindingFlags bindingAttr,
-			[CanBeNull] object binder,
-			[NotNull, ItemNotNull] Type[] types,
-			[CanBeNull] ParameterModifierEx[] modifiers)
-		{
-			if (binder != null) throw new NotImplementedException(nameof(binder));
-			if (types == null) throw new ArgumentNullException(nameof(types));
-			if (types.Any(t => t == null)) throw new ArgumentNullException(nameof(types));
-			if (modifiers != null) throw new NotImplementedException(nameof(modifiers));
-
-			if (bindingAttr == default) bindingAttr = _defaultLookup;
-
-			return type.GetConstructors(bindingAttr).TryFindParametersTypesMatch(types);
-		}
-
-		[CanBeNull]
-		private static T TryFindParametersTypesMatch<T>(
-			[NotNull, ItemNotNull] this IEnumerable<T> methods,
-			[NotNull, ItemNotNull] Type[] types)
-			where T : MethodBase =>
-				methods.Where(
-					m =>
-					{
-						var parameters = m.GetParameters();
-						if (parameters.Length != types.Length) return false;
-
-						for (var i = 0; i < parameters.Length; i++)
-						{
-							if (!parameters[i].ParameterType.Equals(types[i])) return false;
-						}
-
-						return true;
-					})
-					.FirstOrDefault();
-
-		private const BindingFlags _defaultLookup = BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public;
-
-		[MethodImpl(AggressiveInlining)]
-		public static bool GetIsAssignableFrom([NotNull] this Type type, [NotNull] Type otherType) =>
-			type.GetTypeInfo().IsAssignableFrom(otherType.GetTypeInfo());
-
-		[MethodImpl(AggressiveInlining)]
-		public static bool GetIsSubclassOf([NotNull] this Type type, [NotNull] Type c) => type.GetTypeInfo().IsSubclassOf(c);
-
-		[MethodImpl(AggressiveInlining)]
-		public static bool IsDefined([NotNull] this Type type, Type attributeType) =>
-			type.GetTypeInfo().IsDefined(attributeType);
-
-		[MethodImpl(AggressiveInlining)]
-		public static bool IsDefined([NotNull] this Type type, Type attributeType, bool inherit) =>
-			type.GetTypeInfo().IsDefined(attributeType, inherit);
-
-		[MethodImpl(AggressiveInlining)]
-		public static T GetCustomAttribute<T>([NotNull] this Type type) where T : Attribute =>
-			type.GetTypeInfo().GetCustomAttribute<T>();
-
-		[MethodImpl(AggressiveInlining)]
-		public static T GetCustomAttribute<T>([NotNull] this Type type, bool inherit) where T : Attribute =>
-			type.GetTypeInfo().GetCustomAttribute<T>(inherit);
-
-		[NotNull, ItemNotNull]
-		[MethodImpl(AggressiveInlining)]
-		public static Attribute[] GetCustomAttributes([NotNull] this Type type) =>
-			type.GetTypeInfo().GetCustomAttributes().ToArray();
-
-		[NotNull, ItemNotNull]
-		[MethodImpl(AggressiveInlining)]
-		public static Attribute[] GetCustomAttributes([NotNull] this Type type, Type attributeType, bool inherit) =>
-			type.GetTypeInfo().GetCustomAttributes(attributeType, inherit).ToArray();
-#endif
-
 		[MethodImpl(AggressiveInlining)]
 		private static object GetCustomAttributeWithInterfaceSupport(
```

Comment:
Actually these methods are not part of targeting.
Can you move them to Reflection directory where they are used ?

Reply:
This is definitely related to targeting, as all methods eliminate the difference between different targets.

Reply:
GetCustomAttributeWithInterfaceSupport is not a method from .NET SDK.

Reply:
Is this the only such method? So it is the only method that should be taken out, perhaps by making it public  at the same time.